### PR TITLE
Updating socat version support

### PR DESCRIPTION
**Pull request by JacobBarthelmeh**
ZD 18009

### CODE REVIEW

**Comment by JacobBarthelmeh** [comment]
Working on going through socat's `make test` results after the update.

File: tests/api.c [hunk 1]
```diff
@@ -63202,8 +63239,15 @@ static int test_stubs_are_stubs(void)
     CHECKZERO_RET(wolfSSL_CTX_sess_misses, ctx, ctxN);
     CHECKZERO_RET(wolfSSL_CTX_sess_timeouts, ctx, ctxN);
 
+    /* when implemented this should take WOLFSSL object insted, right now
+     * always returns 0 */
+    CHECKZERO_RET(SSL_get_current_expansion, ctx, ctxN);
+
```
Comment (julek-wolfssl) [inline tests/api.c:63245]: Just use `ExpectIntEQ` directly here and pass in `NULL`.

**Review by julek-wolfssl — CHANGES_REQUESTED** [review]

**Comment by julek-wolfssl** [comment]
Obviously would be good to add CI for this.

**Review by julek-wolfssl — CHANGES_REQUESTED** [review]

**Comment by JacobBarthelmeh** [comment]
Retest this please Jenkins

**Review by julek-wolfssl — APPROVED** [review]

**Review by dgarske — APPROVED** [review]

**Pull request merged**